From 4e98f4e07e0cb03eac4b5446909ef84f93dc2ada Mon Sep 17 00:00:00 2001
From: Sofi <sofi+git@mailbox.org>
Date: Mon, 6 Jun 2022 17:14:24 +0200
Subject: [PATCH] Remove `std::process::ExitCode`

Nixpkgs 22.05 does not ship with Rust 1.61 by default.
So the use of ExitCode will be patched out.
---
 src/main.rs | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/src/main.rs b/src/main.rs
index 7b404ae..bdff951 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -12,7 +12,7 @@ use indicatif::ProgressStyle;
 use lazy_static::lazy_static;
 use libium::config::{self, structs::ModIdentifier};
 use octocrab::OctocrabBuilder;
-use std::{process::ExitCode, sync::Arc};
+use std::sync::Arc;
 use subcommands::{add, upgrade};
 use tokio::{runtime, spawn};
 
@@ -32,7 +32,7 @@ lazy_static! {
         .progress_chars("#>-");
 }
 
-fn main() -> ExitCode {
+fn main() {
     let cli = Ferium::parse();
     let mut builder = runtime::Builder::new_multi_thread();
     builder.enable_all();
@@ -44,9 +44,8 @@ fn main() -> ExitCode {
     let runtime = builder.build().expect("Could not initialise Tokio runtime");
     if let Err(err) = runtime.block_on(actual_main(cli)) {
         eprintln!("{}", err.to_string().red().bold());
-        ExitCode::FAILURE
-    } else {
-        ExitCode::SUCCESS
+        runtime.shutdown_background();
+        std::process::exit(1);
     }
 }
 
-- 
2.36.0

